Source/Extensions/Blazorise.DataGrid/DataGrid.razor

```diff
@@ -267,27 +267,39 @@
         }
     };
 
-    protected RenderFragment groupFragment( (int iteration, IEnumerable<DataGridColumn<TItem>> groupableColumns, IEnumerable<IGrouping<object, TItem>> groupedData) x ) => __builder =>
+    protected RenderFragment groupFragment( (int iteration, IEnumerable<DataGridColumn<TItem>> groupableColumns, DataGridColumn<TItem> groupableColumn, IEnumerable<IGrouping<object, TItem>> groupedData) x ) => __builder =>
     {
         @foreach ( var group in x.groupedData )
         {
-            <_DataGridFullColumnSpanRow @key="group.Key.ToString()" TItem="TItem" Columns="@Columns">
-                <Strong>@group.Key.ToString()</Strong>
-            </_DataGridFullColumnSpanRow>
-
-            var nextGroupableColumn = x.groupableColumns?.ElementAtOrDefault( x.iteration );
-            if ( nextGroupableColumn is null )
-            {
-                @foreach ( var item in group )
-                {
-                    @rowFragment( item )
-                }
-            }
-            else
-            {
-                var groupedData = group.GroupBy( x => nextGroupableColumn.GetGroupByFunc().Invoke( x ) );
-                @groupFragment((x.iteration + 1, x.groupableColumns, groupedData))
-            }
+            <TableRowGroup @key="group.Key.ToString()">

```

Comment:
Do we need `.ToString()`?

Reply:
I believe so. The key is an object, we are just expecting that it always is a string. We could cast it sooner somewhere I guess?

Reply:
Make it a string, and convert it to a string when you assign it in a constructor.

Reply:
I guess if we go the state tracking route then we'll have the groups "pre-calcukated" we can cast there.

Reply:
Added state tracking.